Add tiatoolbox_wsi_thumbnailer task

CraigMyles opened the pull request:
This pull request adds a new ToolArena task `tiatoolbox_wsi_thumbnailer` that generates thumbnails from whole-slide images using [TIAToolbox GitHub repository](https://github.com/TissueImageAnalytics/tiatoolbox/) ([Pocock et al., Nature Digital Medicine (2022)](https://www.nature.com/articles/s43856-022-00186-5)).

Given an input directory of WSIs, the tool creates a downsampled thumbnail image (_thumbnail.png) for each slide using TIAToolbox’s WSIReader functionality.

georg-wolflein commented on tasks/tiatoolbox_wsi_thumbnailer/implementation.py:
```diff
@@ -0,0 +1,39 @@
+            # not a supported WSI → skip silently
+            continue
+
+        thumb = reader.slide_thumbnail(resolution=1.25, units="power")
```
Please add `resolution: float = 1.25` and `units: str = "power"` as arguments to the task (modify `task.yaml` as well as this function)

georg-wolflein commented on tasks/tiatoolbox_wsi_thumbnailer/install.sh:
```diff
@@ -0,0 +1,18 @@
+        openslide-tools \
+        libgl1 \
+        && \
+apt-get clean && rm -rf /var/lib/apt/lists/*
```
Nice, keeping image size small:)

georg-wolflein commented on tasks/tiatoolbox_wsi_thumbnailer/data/tcga_read_gdc_manifest.2025-04-26.151110.txt:
Can this file be removed?

georg-wolflein commented on tasks/tiatoolbox_wsi_thumbnailer/task.yaml:
```diff
@@ -0,0 +1,39 @@
+    - source: wsis
+      target: wsis
+test_invocations:
+  - name: small_batch
+    arguments:
+      - name: input_dir
+        value: /mount/input/wsis
+      - name: output_dir
+        value: /mount/output/wsis_thumbs
+    mount:
+      - source: wsis
+        target: wsis
```
Could you please define 3 `test_invocations`? Also please make sure that the test invocations are different from the example invocation.
Some ideas for creating test invocations that are different from the example invocation:
- change the location specified to `input_dir` (and adjust `mount` accordingly)
- invoke with only one file or a subset of files (see example below):
  ```yaml
  - name: one_file
    arguments:
      - name: input_dir
        value: /mount/input/wsis
      - name: output_dir
        value: /mount/output/wsis_thumbs
    mount:
      - source: wsis/TCGA-AG-A011-01Z-00-DX1.155A4093-5EC6-4D38-8CE1-24C045DF0CD8.svs
        target: wsis/TCGA-AG-A011-01Z-00-DX1.155A4093-5EC6-4D38-8CE1-24C045DF0CD8.svs
  - name: subset_of_files
    arguments:
      - name: input_dir
        value: /mount/input/wsis
      - name: output_dir
        value: /mount/output/wsis_thumbs
    mount:
      - source: wsis/TCGA-AG-A011-01Z-00-DX1.155A4093-5EC6-4D38-8CE1-24C045DF0CD8.svs
        target: wsis/TCGA-AG-A011-01Z-00-DX1.155A4093-5EC6-4D38-8CE1-24C045DF0CD8.svs
      - source: wsis/TCGA-DT-5265-01Z-00-DX1.563f09af-8bbe-45cd-9c6d-85a96255e67f.svs
        target: wsis/TCGA-DT-5265-01Z-00-DX1.563f09af-8bbe-45cd-9c6d-85a96255e67f.svs
  ```
  Please also ensure the `tests.py` then covers all the invocations.

georg-wolflein reviewed (changes requested):
Thank you so much for contributing this task! Left a few small requests for changes, and then we should be ready to merge

CraigMyles commented:
Thanks for the suggestions!

I have made the requested updates:
- Added `resolution` and `units` as configurable task parameters.
- Added three distinct test invocations.
- Extended the tests to cover all cases: checking number of output files matches, all outputs are valid PNGs, outputs are non-empty, and ground truth comparison for the full input.

georg-wolflein reviewed (approved):
Thanks!
